File: Cpp/Binary Tree Inorder Traversal.cpp
#include<iostream>
#include<vector>
#include<stack>

using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

vector<int> inorderTraversal(TreeNode *root)
    {
        vector<int> vec;
        stack<TreeNode*> st;
        while(root||!st.empty())
            {
                if(root!=NULL)
                    {
                        st.push(root);
                        root=root->left;
                    }
                else
                    {
                        root=st.top();
                        vec.push_back(root->val);
                        st.pop();
                        root=root->right;
                    }
            }
        return vec;
    }

File: Python/Binary-Tree-Inorder-Traversal.py
#!/usr/bin/env python

# Definition for a  binary tree node
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    # @param root, a tree node
    # @return a list of integers
    def Traversal(self,root,res):
        if root != None:
            self.Traversal(root.left,res);
            res.append(root.val)
            self.Traversal(root.right,res);
        return
    def inorderTraversal(self, root):
        res=[]
        self.Traversal(root,res)
        return res